Detect empty links on markdown rendering and issue an error

zbrox opened the pull request:
Referencing #871

Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [X] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [ ] Have you created/updated the relevant documentation page(s)?

zbrox committed:
Detect empty links on markdown rendering and issue an error

Keats commented on components/rendering/src/markdown.rs:
```diff
@@ -242,6 +242,10 @@ pub fn markdown_to_html(content: &str, context: &RenderContext) -> Result<Render
                         Event::Start(Tag::Image(link_type, src, title))
                     }
+                    Event::Start(Tag::Link(link_type, link, title)) if link.is_empty() => {
+                        error = Some(Error::msg("There is a link that is missing a URL"));
```
Can we add the link type/title to the error message?

zbrox replied:
I actually had it with the title in the message initially but for some reason the `title` variable there is an empty string. I didn't have more time to look into it. My initial guess was that maybe the Markdown parser is not properly parsing a link without a URL.

Keats replied:
ah interesting, I'll see if I can reproduce it and open an issue on pulldown-cmark

zbrox replied:
@Keats did you manage to reproduce it? Is it really an issue or did I misunderstand something?

Keats replied:
I haven't tried yet, I'll check when I have time

Keats replied:
Ah nevermind, `title` is `[I'm an inline-style link with title](https://www.google.com "Google's Homepage")` Google's Homepage. It looks like you don't get the link text in the event

zbrox committed:
Add a test for empty links stopping rendering with an error

zbrox commented on components/rendering/tests/markdown.rs:
```diff
@@ -852,3 +852,24 @@ fn leaves_custom_url_scheme_untouched() {
+    let res = render_content(content, &context);
+
+    assert!(res.is_err());
+}
```
@Keats I was not terribly creative with the test. Should I test for something else other than erring out?

Keats replied:
You can check that the description contains the expected string

zbrox committed:
Assert error message is the expected one

When testing for empty links detection compare the error message
to make sure it's the correct error that stopped the process
and not some unrelated issue.
